models/boards.js

```diff
@@ -10,6 +10,66 @@ Boards.attachSchema(new SimpleSchema({
      */
     type: String,
   },
+  boardKey: {
+    /**
+     * The key of the board
+     */
+    type: String,
+    min: 1,
+    max: 3,
+  },
+  showBoardKey: {
+    /**
+     * Saved configuration how to show the key
+     */
+    allowedValues: [
+      'key-postfix-title-key-bracketed',
+      'key-prefix-title-dash',
+      'key-hide-key',
+      'key-only-key',
+    ],
+    type: String,
+    defaultValue: 'key-hide-key',
+  },
+  fullTitle: {
+    /**
+     * The full title of the board, shown like 'showBoardKey' configured
+     */
+    type: String,
+    autoValue() { // eslint-disable-line consistent-return
+
+      // Fetch required fields
+      const title = this.field('title');
+      const boardKey = this.field('boardKey');
+      const showBoardKey = this.field('showBoardKey');
+
+      // Only act on insert or update
+      // and then all three required fields are set
+      if ((this.isInsert || this.isUpdate) && !this.isSet && title.isSet && boardKey.isSet && showBoardKey.isSet) {
```

Comment:
This "full title" should return a default value, not only returning a value when all the others are set. It should return a value for `this.field('title')` at the very least.